pkg/services/object_manager/placement/netmap.go

```diff
@@ -45,11 +60,28 @@ func (b *netMapBuilder) BuildPlacement(cnr cid.ID, obj *oid.ID, p *netmapSDK.Pla
 	binCnr := make([]byte, sha256.Size)
 	cnr.Encode(binCnr)
 
+	b.mtx.Lock()
+	if nm == b.lastNm {
+		raw, ok := b.containerCache.Get(string(binCnr))
+		b.mtx.Unlock()
+		if ok {
+			cn := raw.(netmapSDK.ContainerNodes)
+			return BuildObjectPlacement(nm, cn, obj)
+		}
+	} else {
+		b.containerCache.Purge()
+		b.mtx.Unlock()
+	}
+
 	cn, err := nm.GetContainerNodes(p, binCnr)
 	if err != nil {
 		return nil, fmt.Errorf("could not get container nodes: %w", err)
 	}
 
+	b.mtx.Lock()
```

Comment:
Can we theoretically build placement for fixed `nm` and `cnr` in case of cache miss? Seems like `GetContainerNodes` isn't synced in this case.

Reply:
Could you elaborate a bit? If cnr is fixed, it is never evicted from the cache.

Reply:
I mean `if nm == b.lastNm && !ok` multiple routines can call `nm.GetContainerNodes`, right? If so, is it expected?

Reply:
Yes: it is a local operation, so nothing is wrong with allowing multiple routines to use it. Also, we do not want to block for different containers.